cpp/src/slice2cpp/Gen.cpp

```diff
@@ -626,6 +626,48 @@ emitOpNameResult(IceUtilInternal::Output& H, const OperationPtr& p, int useWstri
     }
 }
 
+// Returns the client-side result type for an operation - can be void, a single type, or a tuple.
+string
+createResultType(const OperationPtr& p, const string& scope, int useWstring)
+{
+    TypePtr ret = p->returnType();
+    string retS = returnTypeToString(ret, p->returnIsOptional(), scope, p->getMetaData(),
+                                     useWstring | TypeContextCpp11);
+
+    ParamDeclList outParams = p->outParameters();
+
+    if ((outParams.size() > 1) || (ret && outParams.size() > 0))
+    {
+        // Generate a tuple
+        ostringstream os;
+        Output out(os);
+        out << "::std::tuple" << sabrk;
+        if (ret)
+        {
+            out << retS;
+        }
+        for (ParamDeclList::iterator q = outParams.begin(); q != outParams.end(); ++q)
+        {
+            out << typeToString((*q)->type(), (*q)->optional(), scope, (*q)->getMetaData(), useWstring | TypeContextCpp11);
+        }
+        out << eabrk;
+        return os.str();
+    }
+    else if (ret)
+    {
+        return retS;
+    }
+    else if (outParams.size() == 1)
+    {
+        auto q = outParams.begin();
+        return typeToString((*q)->type(), (*q)->optional(), scope, (*q)->getMetaData(), useWstring | TypeContextCpp11);
```

Comment:
Can we just take a reference to the first item?

```
auto& q = outParams.front();
return typeToString(q->type(), q->optional(), scope, q->getMetaData(), useWstring | TypeContextCpp11);
```

Reply:
Fixed